### PR TITLE
response to diagnostic

### PR DESCRIPTION
**Pull request by jdamiron**

### CODE REVIEW

File: diagnostic.md [hunk 1]
```diff
@@ -21,13 +21,17 @@ Write the `curl` request you'd use to retrieve a list of all donuts on the
 server.
 
 ```sh
-# your answer here
+curl --request GET http://www.example.com/donuts
```
Comment (GA-MEB) [inline diagnostic.md:24]: Needs `--include` option to display headers

> "You are expected to use the appropriate HTTP method, and to ask curl to include HTTP headers in its output."

File: diagnostic.md [hunk 2]
```diff
@@ -21,13 +21,17 @@ Write the `curl` request you'd use to retrieve a list of all donuts on the
...
 
 Write an AJAX request to retrieve a list of all donuts on the server.
 
 ```js
-let getDonuts = /* your answer here */;
+let getDonuts = function() {
+  $.ajax({
+    url: 'http://www.example.com/donuts',
+    type: 'GET'
+  });
```
Comment (GA-MEB) [inline diagnostic.md:34]: Don't forget to close the function.

Comment (GA-MEB) [inline diagnostic.md:34]: You're also missing the `.done` and `.fail` handlers.

File: diagnostic.md [hunk 3]
```diff
@@ -36,13 +40,17 @@ Now, we want to get a single donut from the server. Write the `curl` request
 you'd use to retrieve a single donut, using whatever ID you'd like.
 
 ```sh
-# your answer here
+curl --request GET http://www.example.com/donuts/3
```
Comment (GA-MEB) [inline diagnostic.md:43]: `--include`

File: diagnostic.md [hunk 4]
```diff
@@ -36,13 +40,17 @@ Now, we want to get a single donut from the server. Write the `curl` request
...
 
 Write an AJAX request to retrieve a single donut from the server.
 
 ```js
-let getDonut = /* your answer here */;
+let getDonut = function() {
+  $.ajax({
+    url: 'http://www.example.com/donuts/3',
+    type: 'GET'
+  });
```
Comment (GA-MEB) [inline diagnostic.md:53]: As above, you didn't close the function, and you haven't set the `.done` and `.fail` handlers.

File: diagnostic.md [hunk 5]
```diff
@@ -51,13 +59,17 @@ Write the `curl` request you'd use to delete a single donut, using whatever ID
 you'd like.
 
 ```sh
-# your answer here
+curl --request DELETE http://www.example.com/donuts/3
```
Comment (GA-MEB) [inline diagnostic.md:62]: `--include`

File: diagnostic.md [hunk 6]
```diff
@@ -51,13 +59,17 @@ Write the `curl` request you'd use to delete a single donut, using whatever ID
...
 
 Write an AJAX request to delete a single donut from the server.
 
 ```js
-let deleteDonut = /* your answer here */;
+let deleteDonut = function() {
+  $.ajax({
+    url: 'http://www.example.com/donuts/3',
+    type: 'DELETE'
+  });
```
Comment (GA-MEB) [inline diagnostic.md:72]: Ditto to previous. Without handlers, not only can you not operate on data that comes back, you can't tell your program to wait until the request finishes before doing something.

File: diagnostic.md [hunk 7]
```diff
@@ -73,20 +85,38 @@ data in JSON format.
 ```
 
 ```sh
-# your answer here
+curl --request POST http://www.example.com/donuts
```
Comment (MattB-GA-Boston) [inline diagnostic.md]: `--include`

File: diagnostic.md [hunk 8]
```diff
@@ -73,20 +85,38 @@ data in JSON format.
...
+--header "Content-type: application/json" --data '{ "name": "French Cruller",
+"price": "$0.99",
+"done": false,
+"list_id": 3 }'
```
Comment (GA-MEB) [inline diagnostic.md:92]: `--include`

File: diagnostic.md [hunk 9]
```diff
@@ -73,20 +85,38 @@ data in JSON format.
...
 Write an AJAX request to create a single donut on the server using JSON.
 
 ```js
-let createDonut = /* your answer here */;
+let createDonut = function(event) {
+  $.ajax({
+    url: 'http://www.example.com/donuts/3',
+    type: 'POST',
+    data: donut
+  });
```
Comment (GA-MEB) [inline diagnostic.md:103]: In addition to the un-closed function and having no handlers, in this case you're referencing `donut`, which you haven't defined yet.

Comment (GA-MEB) [inline diagnostic.md:103]: In this case, you should be passing in an object that you explicitly define:

```
data: {
  name: 'French Cruller',
  price: '$0.99'
}
```

Comment (GA-MEB) [inline diagnostic.md:103]: Also, this URL is wrong - you don't know yet what ID number the back end will assign to the new object that you're creating. That's why you send things to `../donuts` (no ID) when you create things.

File: diagnostic.md [hunk 10]
```diff
@@ -73,20 +85,38 @@ data in JSON format.
...
+let createDonut = function(event) {
+  event.preventDefault();
+  let donut = new FormData(event.target);
+  $.ajax({
+    url: 'http://www.example.com/donuts/3',
+    type: 'POST',
+    contentType: false,
+    processData: false,
+    data: donut
+  });
```
Comment (GA-MEB) [inline diagnostic.md:119]: Missing `.done` and `.fail` handlers, and wrong URL (see previous note).

File: diagnostic.md [hunk 11]
```diff
@@ -105,18 +135,36 @@ JSON.
 ```
 
 ```sh
-# your answer here
+curl --request PATCH http://www.example.com/donuts/42 \
+--header "Content-Type: application/json" \
+--data '{
+"name": "Krüller",
+"done": true
```
Comment (GA-MEB) [inline diagnostic.md:142]: Not sure why there's a `done` property on the data here.

Comment (GA-MEB) [inline diagnostic.md:142]: `--include`

File: diagnostic.md [hunk 12]
```diff
@@ -105,18 +135,36 @@ JSON.
...
 
 Write an AJAX request to change the donut on the server using JSON.
 
 ```js
-let changeDonut = /* your answer here */;
+let changeDonut = function() {
+  $.ajax({
+    url: 'http://www.example.com/donuts/42',
+    type: 'PATCH'                   
+  });
```
Comment (GA-MEB) [inline diagnostic.md:153]: Where is the data for the PATCH request to send?

Comment (GA-MEB) [inline diagnostic.md:153]: Again, no handlers.

File: diagnostic.md [hunk 13]
```diff
@@ -105,18 +135,36 @@ JSON.
...
+let changeDonut = function(event) {
+  event.preventDefault();
+  let donut = new FormData(event.target);
+  $.ajax({
+    url: 'http://www.example.com/donuts/42',
+    type: 'PATCH',
+    contentType: false,
+    processData: false,
+    data: donut
+  };
```
Comment (GA-MEB) [inline diagnostic.md:169]: See previous comments about handlers; otherwise, this is right.

**Comment by GA-MEB** [comment]
5.75 / 12
(-0.25 on Q1, Q3, Q5, Q7, Q10 for `--include`; -0.5 on Q2, Q4, Q6, Q8, Q9, Q11, Q12 for missing handlers and not closing your function; -0.25 on Q8 and Q9 for wrong URL; -0.25 on Q8 and Q11 for not defining your data.)

I know this score looks bad, but it's mostly because you made one significant mistake (not closing your function & forgetting to add handlers to your AJAX) on more than half of the questions. If you fixed that one thing, you'd be getting a 9.25 / 12.

**Pull request closed without merging**